src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs

```diff
@@ -219,7 +217,42 @@ public static void Initialize()
         }
     }
 
-#if NET
+    private static void TryInitializeContinuousProfiling()
+    {
+#if NETFRAMEWORK
+        // Check AppDomain FIRST, before touching the flag
+        if (!AppDomain.CurrentDomain.IsDefaultAppDomain())
```

Comment:
I beleive on ASP.NET app we do not inited in default app domain, so that would not be executed - and we would not have canary thread.